### PR TITLE
[feat] Add Mysql docker container

### PR DESCRIPTION
**Pull request by EpicPi**
## Changes made
Add mysql docker container.

To run dev environment after this pr lands:
1. bin/dev pull
2. bin/dev py-reqs
3. bin/dev up

## Testing
- [ x] End-to-End

### CODE REVIEW

File: bin/dev [hunk 1]
```diff
@@ -18,7 +18,7 @@ findBackendContainerId(){
 # args: commands to run in container
 runDockerExec(){
     findBackendContainerId container_id
-    cmd="docker exec -it -u $(id -u):$(id -g) $container_id "$@
+    cmd="docker exec -it -u $(id -u):$(id -g) -w /home/worker/app/server $container_id "$@
```
Comment (EpicPi) [inline bin/dev:21]: Bin/dev commands are kindof broken - working on a fix in another pr, temporary solution here.

File: bin/infra [hunk 2]
```diff
@@ -39,12 +39,13 @@ dockerPush(){
 
 # Pull image from aws ecr
 dockerPull(){
+    dockerLogin
```
Comment (EpicPi) [inline bin/infra:42]: Sometimes you need to login before you can pull.

File: bin/infra [hunk 3]
```diff
@@ -39,12 +39,13 @@ dockerPush(){
...
     bin/util logAndRun "docker pull $DEV_ECR_REPO"
 }
 
 # Build dev image
 dockerBuild(){
-    bin/util logAndRun "docker build --pull --rm -t $DEV_ECR_REPO - < Dockerfile_dev"
+    bin/util logAndRun "docker build --target dev --pull --rm -t $DEV_ECR_REPO - < Dockerfile"
```
Comment (EpicPi) [inline bin/infra:48]: leftover from Dockerfile consolidation a couple prs ago

File: docker-compose.yml [hunk 4]
```diff
@@ -18,13 +18,30 @@ services:
     ports:
       - 27017:27017
 
+  mysql:
+    image: mysql:8.0.23
```
Comment (EpicPi) [inline docker-compose.yml:22]: version used in aws

File: server/src/clients/mysql.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import os
+
+import pymysql  # type: ignore
```
Comment (EpicPi) [inline server/src/clients/mysql.py:3]: mypy hates me. I gave up - plz take a look and see if you can fix when adding the sqlalchemy models @SammyBaek

File: .github/workflows/devDockerBuild.yml [hunk 6]
```diff
@@ -13,12 +13,12 @@ jobs:
...
       - uses: actions/checkout@v2
 
-      - name: Check If Dockerfile_dev has changed
+      - name: Check If Dockerfile has changed
         uses: technote-space/get-diff-action@v5
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          PATTERNS: Dockerfile_dev
+          PATTERNS: Dockerfile
```
Comment (EpicPi) [inline .github/workflows/devDockerBuild.yml:21]: leftover from Dockerfile consolidation a couple prs ago

File: bin/dev (hunk 1 above)
Comment (SammyBaek) [reply bin/dev:21]: Did this work? Nick tried bunch of things, so I wasn't sure what was the fix. I think the final fixture was running `pre-commit` locally outside of docker to set everything up.

File: docker-compose.yml (hunk 4 above)
Comment (SammyBaek) [reply docker-compose.yml:22]: does this change at all over time? or does it not matter

File: server/src/clients/mysql.py (hunk 5 above)
Comment (SammyBaek) [reply server/src/clients/mysql.py:3]: lol ignore in import is more than fine :') what exactly is the issue?

File: server/src/clients/mysql.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+# Test class for connecting to mysql database
+
+mysql_user = os.environ.get("MYSQL_USER") or ""
```
Comment (SammyBaek) [inline server/src/clients/mysql.py:7]:
```suggestion
mysql_user = os.getenv("MYSQL_USER", "")
```

**Review by SammyBaek — APPROVED** [review]

File: docker-compose.yml (hunk 4 above)
Comment (EpicPi) [reply docker-compose.yml:22]: it does get updates - we'll update it manually as the aws one updates

File: server/src/clients/mysql.py (hunk 5 above)
Comment (EpicPi) [reply server/src/clients/mysql.py:3]: Mypy being a pain. We can try to clean it up sometime later.

File: bin/dev (hunk 1 above)
Comment (EpicPi) [reply bin/dev:21]: interesting. This is necessary for other commands to work. Didn't consider nick's problem when doing this.

**Pull request merged**